pos_tare/static/src/js/screens.js

```diff
@@ -18,7 +20,21 @@ odoo.define('pos_tare.screens', function (require) {
                 var order = this.pos.get_order();
                 var selected_order_line = order.get_selected_orderline();
                 var tare_weight = code.value;
-                selected_order_line.set_tare(tare_weight, true);
+                var tare_uom = this.pos.config.iface_tare_uom_id[0];
+                var tare_unit = this.pos.units_by_id[tare_uom];
+                var kilogram_uom = this.pos.units.filter(
+                    function (u) {
+                        return u.name === "kg";
```

Comment:
sorry. I still don't understand why there are hardcoded text regarding uom in this module. there is no way to have something more modular ? if it's not possible, do you think you could write a section in the ROADMAP section, to mention to do it (maybe when odoo PoS will be refactored...)

Reply:
Assume you do not hard code this UOM (the one that is used to encode the tare barcode labels). Let's say this UOM is configured to be kilogram. Now print a tare label with 1 unit of UOM, that's one kilogram. Assume the user changes the configuration for this UOM and selects "metric ton". Afterwards one scans the old tare barcode label, she'll get a tare of one metric ton instead of the correct weight of the container that is one kilogram. If you have a better solution than hard coding, I'm happy to change that.

Reply:
Hope this is clear enough...

Reply:
@legalsylvain What's your opinion on that? Are we all good?

Reply:
Sorry. Back on that point. I was fully busy the last monthes. 

I think _" Assume the user changes the configuration for this UOM and selects "metric ton"_ can not occure.

I tested your Use case with demo data in V12 : 
- select the product with kilogram unit (Cable Management Box) (``stock.product_cable_management_box``)
- change the quantity (70kg in hand).
- select the uom kg (``uom.product_uom_kgm``)
- try to change uom.

You'll have this errors.

![image](https://user-images.githubusercontent.com/3407482/106439753-5764a980-6478-11eb-8a8f-63014a66d5ca.png)

My opinion is that odoo doesn't handle that use case. (that is quite hard to handle, I think). so, you should not have to handle that use case in a OCA module, if Odoo core module doesn't allow it.
It will make the code simpler.
Dont you think ?